src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetAddressesForPersonService.kt

```diff
@@ -0,0 +1,20 @@
+package uk.gov.justice.digital.hmpps.hmppsintegrationapi.services
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.gateways.NomisGateway
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.gateways.ProbationOffenderSearchGateway
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.Address
+
+@Service
+class GetAddressesForPersonService(
+  @Autowired val probationOffenderSearchGateway: ProbationOffenderSearchGateway,
+  @Autowired val nomisGateway: NomisGateway
+) {
+  fun execute(id: String): List<Address> {
+    val addressesFromProbationOffenderSearch = probationOffenderSearchGateway.getAddressesForPerson(id)
+    val addressesFromNomis = nomisGateway.getAddressesForPerson(id)
+
+    return addressesFromProbationOffenderSearch + addressesFromNomis
```

Comment:
`return addressesFromProbationOffenderSearch + addressesFromNomis`

We did the exact same thing in the search endpoint for this. I feel like this could be an area of optimisation in the future when we need to pull details from more than just a couple of systems. Again, just a side-note. No need for action :)

Reply:
Agreedaroo, one day I can imagine we'll want to sort them or something but for now like you said I think we're good.